metals/src/main/scala/scala/meta/internal/metals/Trace.scala

```diff
@@ -4,45 +4,78 @@ import java.io.PrintWriter
 import java.nio.file.Files
 import java.nio.file.StandardOpenOption
 
+import scala.annotation.tailrec
+import scala.util.Try
+
 import scala.meta.internal.io.PathIO
 import scala.meta.io.AbsolutePath
 
+import dev.dirs.ProjectDirectories
+
 /**
  * Manages JSON-RPC tracing of incoming/outgoing messages via BSP and LSP.
  */
 object Trace {
-  val metalsLog: AbsolutePath =
-    PathIO.workingDirectory.resolve(".metals/metals.log")
+  // jvm-directories can fail for less common OS versions: https://github.com/soc/directories-jvm/issues/17
+  val globalDirectory: Option[AbsolutePath] =
+    Try {
+      val projectDirectories =
+        ProjectDirectories.from("org", "scalameta", "metals")
+      // NOTE(olafur): strictly speaking we should use `dataDir` instead of `cacheDir` but on
+      // macOS this maps to `$HOME/Library/Application Support` which has an annoying space in
+      // the path making it difficult to tail/cat from the terminal and cmd+click from VS Code.
+      // Instead, we use the `cacheDir` which has no spaces. The logs are safe to remove so
+      // putting them in the "cache directory" makes more sense compared to the "config directory".
+      AbsolutePath(projectDirectories.cacheDir)
+    }.toOption
+
+  val localDirectory: AbsolutePath = PathIO.workingDirectory.resolve(".metals/")
+  val metalsLog: AbsolutePath = localDirectory.resolve("metals.log")
 
   def protocolTracePath(
       protocolName: String,
-      workspace: AbsolutePath = PathIO.workingDirectory
+      workspace: AbsolutePath = localDirectory
   ): AbsolutePath = {
     val traceFilename = s"${protocolName.toLowerCase}.trace.json"
-    workspace.resolve(".metals").resolve(traceFilename)
+    workspace.resolve(traceFilename)
   }
 
+  /**
+   * Setups trace printer for a given protocol name and workspace.
+   * Searches for trace file in provided workspace. If there is no such file in the workspace, fall backs to the global directory.
+   */
   def setupTracePrinter(
       protocolName: String,
-      workspace: AbsolutePath
+      workspace: AbsolutePath = localDirectory
   ): Option[PrintWriter] = {
-    val tracePath = protocolTracePath(protocolName, workspace)
-    val path = tracePath.toString()
-    if (tracePath.isFile) {
-      scribe.info(s"tracing is enabled: $path")
-      val fos = Files.newOutputStream(
-        tracePath.toNIO,
-        StandardOpenOption.CREATE,
-        StandardOpenOption.TRUNCATE_EXISTING // don't append infinitely to existing file
-      )
-      Some(new PrintWriter(fos))
-    } else {
-      scribe.info(
-        s"tracing is disabled for protocol $protocolName, to enable tracing of incoming " +
-          s"and outgoing JSON messages create an empty file at $path"
-      )
-      None
-    }
-  }
+    val metalsDir =
+      if (workspace.toNIO.endsWith(".metals")) workspace
```

Comment:
Why would the workspace be the `.metals` directory?

Reply:
That's because previously as a default I used `PathIO.workingDirectory.resolve(".metals/")` which makes no sense because it isn't a workspace dir. Now I default to the `PathIO.workingDirectory` and problem is gone